src/crimson/osd/pg.cc

```diff
@@ -680,7 +680,9 @@ PG::get_locked_obc(
 {
   return get_or_load_head_obc(oid.get_head()).safe_then(
     [this, op, oid, type](auto p) -> load_obc_ertr::future<ObjectContextRef>{
-      auto &[head_obc, head_existed] = p;
+      crimson::osd::ObjectContextRef head_obc;
+      bool head_existed;
+      std::tie(head_obc, head_existed) = p;
```

Comment:
i think we can apply the same technique to other places where we are capturing the structured binding "variables".

Reply:
```diff
diff --git a/src/crimson/osd/pg.cc b/src/crimson/osd/pg.cc
index f66b7cacdb..a70d867047 100644
--- a/src/crimson/osd/pg.cc
+++ b/src/crimson/osd/pg.cc
@@ -683,7 +683,7 @@ PG::get_locked_obc(
       auto &[head_obc, head_existed] = p;
       if (oid.is_head()) {
    if (head_existed) {
-     return head_obc->get_lock_type(op, type).then([head_obc] {
+     return head_obc->get_lock_type(op, type).then([head_obc=head_obc] {
        ceph_assert(head_obc->loaded);
        return load_obc_ertr::make_ready_future<ObjectContextRef>(head_obc);
      });
```

Reply:
Fine w me. I have verified that the code produced
is the same on both Clang and gcc. So it's a matter of taste. 

I will update the PR to this format.

Reply:
i think the reason i like my "style" is that it is simpler: without introducing putting the type of the captured variable explicitly and assign their value using "tie".